holoplot/__init__.py

```diff
@@ -0,0 +1,385 @@
+import holoviews
+from holoviews import (
+    extension as _extension, Dimensioned as _Dimensioned, HoloMap as _HoloMap
+)
+
+from bokeh.io import export_png as _export_png, show as _show, save as _save
+from bokeh.resources import CDN as _CDN
+
+from .converter import HoloViewsConverter
+
+renderer = holoviews.renderer('bokeh')
+
+# Register plotting interfaces
+def _patch_plot(self):
+    return HoloPlot(self)
+
+
+def patch(library, extension=None):
+    """
+    Patch library to support HoloViews based plotting API.
+    """
+    if not isinstance(library, list): library = [library]
+    if 'streamz' in library:
+        try:
+            import streamz.dataframe as sdf
+        except ImportError:
+            raise ImportError('Could not patch plotting API onto streamz. '
+                              'Streamz could not be imported.')
+        sdf.DataFrame.plot = property(_patch_plot)
+        sdf.DataFrames.plot = property(_patch_plot)
+        sdf.Series.plot = property(_patch_plot)
+        sdf.Seriess.plot = property(_patch_plot)
+    if 'pandas' in library:
+        try:
+            import pandas as pd
+        except:
+            raise ImportError('Could not patch plotting API onto pandas. '
+                              'Pandas could not be imported.')
+        pd.DataFrame.plot = property(_patch_plot)
+        pd.Series.plot = property(_patch_plot)
+    if 'dask' in library:
+        try:
+            import dask.dataframe as dd
+        except:
+            raise ImportError('Could not patch plotting API onto dask. '
+                              'Dask could not be imported.')
+        dd.DataFrame.plot = property(_patch_plot)
+        dd.Series.plot = property(_patch_plot)
+    if 'intake' in library:
+        try:
+            from intake.source.base import DataSource
+        except ImportError:
+            raise ImportError('Could not patch plotting API onto intake. '
+                              'Intake could not be imported.')
+        DataSource.plot = property(_patch_plot)
+    if 'xarray' in library:
+        try:
+            from xarray import DataArray
+        except ImportError:
+            raise ImportError('Could not patch plotting API onto xarray. '
+                              'Xarray could not be imported.')
+        DataArray.plot = property(_patch_plot)
+    if extension:
+        _extension(extension)
+
+
+class HoloPlot(object):
+
+    def __init__(self, data):
+        self._data = data
+
+    def __call__(self, x=None, y=None, kind=None, backlog=1000,
+             width=700, height=300, title=None, grid=False,
+             legend=True, logx=False, logy=False, loglog=False,
+             xticks=None, yticks=None, xlim=None, ylim=None, rot=None,
+             fontsize=None, colormap=None, hover=True, **kwds):
```

Comment:
Is there a specific reason these are all explicit keyword arguments, instead of Parameters or part of `kwds` so that they can be checked against a lookup table?  Is it for tab-completion?